File: Graph/797. All Paths From Source to Target/797.java
import java.util.ArrayList;

class Solution {
    public List<List<Integer>> ans = new ArrayList<>();

    public void go(int start, List<Integer> lis, int end, int[][] graph){
        if(start == end){
            lis.add(end);
            ans.add(new ArrayList<>(lis));
            lis.remove(lis.size()-1);
            return;
        }

        lis.add(start);
        for(int des : graph[start]){
            go(des, lis, end, graph);
        }
        lis.remove(lis.size()-1);

    }

    public List<List<Integer>> allPathsSourceTarget(int[][] graph) {
        int end = graph.length - 1;
        go(0, new ArrayList<>(), end, graph);
        return ans;
    }
}

File: 13-Graph/DFS/797. All Paths From Source to Target/797.py
class Solution:
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        end = len(graph) - 1
        ans = []

        def go(start, lis):
            if start == end:
                ans.append(lis+[start])
                return
            
            lis.append(start)
            for des in graph[start]:
                go(des, lis)
            lis.pop()

        go(0,[])

        return ans
